moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -251,13 +251,13 @@ bool CurrentStateMonitor::waitForCurrentState(const rclcpp::Time& t, double wait
 
 bool CurrentStateMonitor::waitForCompleteState(double wait_time) const
 {
-  double slept_time = 0.0;
-  double sleep_step_s = std::min(0.05, wait_time / 10.0);
-  rclcpp::Duration sleep_step(sleep_step_s);
-  while (!haveCompleteState() && slept_time < wait_time)
+  const double sleep_step_s = std::min(0.05, wait_time / 10.0);
+  const auto sleep_step_duration = rclcpp::Duration::from_seconds(sleep_step_s * 1e9);
+  double slept_time_s = 0.0;
+  while (!haveCompleteState() && slept_time_s < wait_time)
   {
-    rclcpp::sleep_for(sleep_step.to_chrono<std::chrono::nanoseconds>());
-    slept_time += sleep_step_s;
+    rclcpp::sleep_for(sleep_step_duration.to_chrono<std::chrono::nanoseconds>());
+    slept_time_s += sleep_step_s;
   }
   return haveCompleteState();
 }
```

Comment:
What happens with the old logic (on rolling):

wait_time = 10 (seconds)
slept_time = 0.0
sleep_time_s = 1
sleep_step = 1ns (constructor for double seconds is gone and the single argument constructor for Duration is now nanoseconds, there is a compiler warning that prints saying this is deprecated)

sleeps for 10 1ns amounts and returns...

As a result servo never hasCompleteState and calls `exit(EXIT_FAILURE);` in the constructor of Servo.  I moved the wait for complete state into the start function.

As an aside I really wish I had colorized ros2 output.  Has anyone figured out how to get that?  It would have been much quicker to debug if the error it printed was red.

Reply:
> As an aside I really wish I had colorized ros2 output. Has anyone figured out how to get that? It would have been much quicker to debug if the error it printed was red.

You could do it with `export RCUTILS_COLORIZED_OUTPUT=1`

Reply:
^Thanks @JafarAbdi! Adding to my bashrc